docs/AUTH_PROTOCOL.md

```diff
@@ -0,0 +1,393 @@
+        - `fallbackMessage`: The message to display if the UI doesn't know how to handle the `reason`
+    - Response: `response` event with arguments:
+        - `newPassword`: The user's new password
+
+- `pin`: UI for entering a numeric PIN code
+    - This method is always primary
+    - Arguments:
+        - `length`: The number of digits in the PIN code
+        - `overridePrompt`: Overrides the default prompt (or, more likely, adds a prompt) to the PIN entry UI. Optional. Should only be used if absolutely necessary
+    - Response: `reponse` event with arguments:
+        - `pin`: The PIN code entered by the user
+
+- `otp`: UI for entering a one-time-password
+    - This method is always primary
+    - Arguments:
+        - `alphanumeric`: Allow alphabetic characters in the OTP. Defaults to false
+        - `length`: The number of digits/characters in the OTP
+        - `prefix`: Some OTPs start with a prefix (e.g. Google likes to use `G-######` as their format). Defaults to empty
+        - `suffix`: Similar to `prefix`, but for a suffix instead
+        - `overridePrompt`: Overrides the default prompt (or, more likely, adds a prompt) to the OTP entry UI. Optional. Should only be used if absolutely necessary
+    - Response: `response` event with arguments:
+        - `otp`: The OTP entered by the user. Neither the prefix nor the suffix are included.
+
+- `chooser`: UI for choosing from a list of options
+    - This method is always primary
+    - Arguments:
+        - `prompt`: A string to display to the user, explaining what they're choosing between. Optional.
+        - `options`: An array of objects that describes the options that the user can select between. Each object in the array has the following keys:
+            - `key`: The machine-usable string that identifies this option
+            - `name`: The display name for this option, to be presented to the user
+    - Response: `response` event with arguments:
+        - `selection`: The selected option's `key`
+
+- `eidp`: Web-based SSO login
+    - The frontend displays a QR code and/or an embedded web browser that loads a web page for single-sign-on
+    - This method is always primary
+    - Arguments:
+        - `url`: The URL of the SSO login page. The QR code encodes this URL, and the web browser navigates to the URL.
+        - `displayUrl`: A simplified URL to display to the user in text, suitible for the user to type manually. Optional
+        - `code`: A short code that is displayed to the user along with the `displayUrl`. This allows the user to complete authentication from an alternative device (like their phone)
+        - `showDone`: If true, the authenticator is unable to detect on its own when authentication has succeeded and the frontend should have a "Done" button for the user to press when they are done
+        - `embedded`: Allow the use of an embedded web view directly on the login screen. Defaults to true
```

Comment:
This is not as simple as this IMHO. A embedded webview requires an other object of options to control it and ensure it behaves, for example:

```
urisRegex: regex of allowed URIs the user can navigate to (you want to allow to open the page conditions or privacy policy, but not to go anywhere else)
navigationButtons: Wehter to show them or not
navigationBar: whether to see it (in read-only or not)
sessionBrowserAllowed: if enabled
```

And probably more, since we need to launch a subprocess that must follow such parameters, so that's likely just another sub-protocol per se :)